src/trix/models/composition.coffee

```diff
@@ -78,16 +78,22 @@ class Trix.Composition extends Trix.BasicObject
     block = document.getBlockAtIndex(index)
 
     if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
+      if block.text.getStringAtRange([offset - 1, offset]) is "\n"
+        document = document.removeTextAtRange([position - 1, position])
+      else if offset - 1 isnt 0
+        position += 1
```

Comment:
When `breakFormattedText` was called, the last character in a block containing a bullet and a header would get removed (causing [this test](https://github.com/basecamp/trix/pull/263/files#diff-c6826d818993c65553b46070e3a0efbeR298) to fail).

The change prevents this from happening by incrementing the position (instead of removing the next character) when the next character is **not** a newline.